l1-contracts/contracts/state-transition/chain-deps/facets/Executor.sol

```diff
@@ -243,8 +260,82 @@ contract ExecutorFacet is ZKChainBase, IExecutor {
                 if (_expectedSystemContractUpgradeTxHash != logValue) {
                     revert TxHashMismatch();
                 }
-            } else if (logKey > uint256(SystemLogKey.EXPECTED_SYSTEM_CONTRACT_UPGRADE_TX_HASH_KEY)) {
-                revert UnexpectedSystemLog(logKey);
+            } else if (
+                logKey > uint256(SystemLogKey.EXPECTED_SYSTEM_CONTRACT_UPGRADE_TX_HASH_KEY) && logKey <= MAX_LOG_KEY
+            ) {
+                // kl todo add only non L1 check here
+                ViaIR memory viaIR;
+                // we are using the fact that msg_root_logs are emitted after each other, starting with the chainId.
+                // Extract the values to be compared to/used such as the log sender, key, and value
+                // slither-disable-next-line unused-return
+                i = i.uncheckedAdd(L2_TO_L1_LOG_SERIALIZE_SIZE);
```

Comment:
unless it severely affects performance, it is better to clean this up to use an array / remove code repetition + explain why exactly 4 times do we read keys/values